realm/realm-library/src/main/java/io/realm/RealmConfiguration.java

```diff
@@ -515,6 +554,36 @@ public Builder initialData(Realm.Transaction transaction) {
             return this;
         }
 
+        /**
+         * Copies the Realm file from the given asset file path.
+         *
+         * When opening the Realm for the first time, instead of creating an empty file,
+         * the Realm file will be copied from the provided assets file and used instead.
+         * WARNING: This could potentially be a lengthy operation so should ideally be done on a background thread.
+         *
+         * @param context Android application context.
+         * @param assetFile path to the asset database file.
+         */
+        public Builder assetFile(Context context, final String assetFile) {
+            if (context == null) {
+                throw new IllegalArgumentException("A non-null Context must be provided");
+            }
+            if (TextUtils.isEmpty(assetFile)) {
+                throw new IllegalArgumentException("A non-empty asset file path must be provided");
+            }
+            if (new File(folder, fileName).exists()) {
```

Comment:
This check here seems kinda harsh. It prevents the use case where you might delete the file _after_ the configuration was created. I would just remove it, since this method is designed to "fail" silently anyway. There is also the check in `copyRealmFileIfNeeded` that prevents the copy anyway.